### DISCUSSION TITLE
Sending requests is pending until previous request done

### DISCUSSION THREAD

**Discussion by sonientaegi** [axios/axios#5867] [Q&A] [answered]
### Describe the issue

When I make multiple request using axios ( continous enter or click ) they are queued immediately and script runs well. But actual http requests are pending until previous request responses.

My service takes several seconds to make response. So I have expected axios tosses them to secuce all and “.then()” continuously - Surely, sequence must not be concerned due asynchronous request.  

Is there any way to make every proceeding request not to wait until getting response of previous request? My remote service runs on flask with threaded=True. Sever responses as I expected when I make multiple enter key on url bar, using axios on multiple “different kind of” browsers - not even  multiple “same” browsers.

### Example Code

```js
pseudo.

// client
Service extends Component {
  onSubmit() {
    axios.get(“/api”)
      .then(r => {
        // blahh
    });
  }

  render() {
    return(
      <Button onClick{this.onSubmit}
 />
    );
  }
}

// server
@blah.route(“/api”)
def process(…)
  # IN
  resp = SomeLongProcesses()
  return resp
  # OUT
}
```

### Expected behavior

IN
IN
IN
IN
…
OUT <- Begin of response.
IN
OUT
OUT
OUT
OUT
IN
OUT
…

**Comment by akin-tekeoglu** [comment]
This is related to the max number of concurrent connections in a browser. E.g you can make 6 HTTP requests at any given time in Chrome.

https://stackoverflow.com/questions/985431/max-parallel-http-connections-in-a-browser
https://ishwar-rimal.medium.com/why-does-your-browser-limit-the-number-of-concurrent-network-calls-1ae5d50863dd

**Comment by yasuf** [comment, marked as answer]
can you verify your server can take multiple requests in parallel via `curl` commands? also can you post the network tab when you click multiple times on the button? to see the requests being made and the time it takes for each to complete
